cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java

```diff
@@ -446,6 +442,33 @@ public void deletePublicIps(AzureClient azureClient, String resourceGroupName, C
         }
     }
 
+    @Retryable(backoff = @Backoff(delay = 1000, multiplier = 2, maxDelay = 10000), maxAttempts = 5)
+    public void deleteLoadBalancers(AzureClient azureClient, String resourceGroupName, Collection<String> loadBalancerNames) {
+        LOGGER.info("Deleting load balancers: {}", loadBalancerNames);
+        try {
+            List<Completable> deleteCompletables = new ArrayList<>();
+            for (String loadBalancerName : loadBalancerNames) {
+                deleteCompletables.add(azureClient.deleteLoadBalancerAsync(resourceGroupName, loadBalancerName)
+                        .doOnError(throwable -> {
+                            LOGGER.error("Error happened on azure load balancer delete: {}", loadBalancerName, throwable);
+                        })
+                        .subscribeOn(Schedulers.io()));
+            }
+            Completable.mergeDelayError(deleteCompletables)
+                    .doOnCompleted(() -> {
+                        LOGGER.debug("Azure load balancers successfully deleted.");
+                    })
+                    .await();
+        } catch (CompositeException e) {
+            String errorMessages = e.getExceptions().stream().map(Throwable::getMessage).collect(Collectors.joining());
+            LOGGER.error("Error(s) occurred while waiting for load balancer deletion: {}", errorMessages);
+            throw new CloudbreakServiceException("Error(s) occurred while waiting for load balancer deletion: " + errorMessages, e);
```

Comment:
I'm not sure the reduction in code is worth the additional indirection:
Exception catch:
```java
} catch (CompositeException e) {
    logAndWrapCompositeException(e, "Error(s) occurred while waiting for load balancer deletion: {}");
}
```

Method definition:
```java
private void logAndWrapCompositeException(CompositeException e, String logMessage) {
    String errorMessages = e.getExceptions().stream().map(Throwable::getMessage).collect(Collectors.joining());
    LOGGER.error(logMessage, errorMessages);
    throw new CloudbreakServiceException(logMessage + errorMessages, e);
}
```

The inline flat map of exceptions is a bit much, but otherwise it's just a Log statement and an exception wrap.

Reply:
I would do it, as it would replace 6 duplicates

Reply:
Done, please see `logAndWrapCompositeException` at the bottom of `AzureUtils`.